tests/integration/main.go

```diff
@@ -131,9 +134,14 @@ func (tr *TestRun) runStep(step Step, verbose bool) {
 	}
 }
 
-// Starts docker container and sequentially runs steps
-func (tr *TestRun) ExecuteSteps(wg *sync.WaitGroup, steps []Step) {
+// ExecuteStepsInWaitGroup runs steps in a WaitGroup so that the test run can be run in parallel.
+func (tr *TestRun) ExecuteStepsInWaitGroup(wg *sync.WaitGroup, steps []Step) {
 	defer wg.Done()
+	tr.ExecuteSteps(steps)
```

Comment:
Shouldn't we spawn a goroutine here or at the call site? Depends exactly how you're intending to design your concurrency tho

Reply:
Added option to run tests in parallel or in sequence using `--parallel` flag